AEROGEAR-9175 - Creating a voyager server container image for use with OpenShift (#689)
aerogear/mobile-docs: modules/ROOT/pages/_partials/sync/creating-a-voyager-server-dockerfile-for-use-with-openshift.adoc

> [id="creating-a-voyager-server-dockerfile-for-use-with-openshift-{context}"]

Reviewer: ```suggestion
[id="creating-a-voyager-server-container-image-for-use-with-openshift-{context}"]
```

> = Creating a voyager server dockerfile for use with OpenShift

Reviewer: ```suggestion
= Creating a voyager server container image for use with OpenShift
```

> This section describes how to create a dockerfile using a local voyager server repo.

Reviewer: ```suggestion
This section describes how to create a container image using a local voyager server repo.
```

Although it is not covered in this section, you can add the Apollo Voyager Client.
For more information, see xref:server-getting-started-with-an-example-data-source-{context}[Getting started with an example data source].

.Prerequisites

* You have cloned the link:https://github.com/aerogear/voyager-server[Apollo Voyager Server] locally.

> * Docker is installed and is running on your local machine. Use `sudo systemctl start docker` to start docker.

Reviewer: Alternatively podman, or buildah

.Procedure

. Open a terminal and browse to the Apollo Voyager Server repo on your local machine.
+

> . Create a Docker file, for example, `touch Dockerfile`.

Reviewer: ```suggestion
. Create a Dockerfile, for example, `touch Dockerfile`.
```

+

> . Open the Docker file and add the following content, then save the changes:

Reviewer: ```suggestion
. Open the Dockerfile and add the following content, then save the changes:
```

+
[source,dockerfile]
----
FROM node:8
WORKDIR /usr/src/app
# copy Node.js specific files
COPY package*.json ./
# copy application source file to the workdir
COPY index.js .
RUN npm install
# TCP port that application is listening on
EXPOSE 4000
CMD [ "node", "index.js" ]
----
+

> . Build the Docker container and tag it:

Reviewer: ```suggestion
. Build the container image and tag it:
```

+
[source,bash]
----
$ docker build -t aerogear/showcase-server:latest
----
+

> NOTE: Alternatively, you can use `Podman` to build a Docker container: `podman build -t aerogear/showcase-server:latest .`

Reviewer: ```suggestion
NOTE: Alternatively, you can use `Podman` to build a container image: `podman build -t aerogear/showcase-server:latest .`
```

.Verification steps

For the example above, the output should be:
[source,bash]
----
$ Successfully built 6490624e298b
$ Successfully tagged aerogear/showcase-server:latest
----

.Additional resources

> * For more information about Docker, see link:https://docs.docker.com/compose/gettingstarted/[Get started with Docker Compose].

Reviewer: ```suggestion
```

[id="creating-a-voyager-server-dockerfile-for-use-with-openshift-{context}"]
= Creating a voyager server container image for use with OpenShift

This section describes how to create a container image using a local voyager server repo.

> Although it is not covered in this section, you can add the Apollo Voyager Client.

Reviewer 2: why would you do this?
and how?

suggestion: just remove

For more information, see xref:server-getting-started-with-an-example-data-source-{context}[Getting started with an example data source].

.Prerequisites

* You have cloned the link:https://github.com/aerogear/voyager-server[Apollo Voyager Server] locally.

> * Docker is installed and is running on your local machine. Use `sudo systemctl start docker` to start docker. As an alternative, use `podman`, or `buildah`.

Reviewer: That `sudo systemctl start docker` -- what OS are we assuming here?

- MacOS & Windows: don't use systemd, so it doesn't apply
- Fedora: now defaults to `podman` instead of docker on all supported versions (I think)
- RHEL: RHEL8 defaults to `podman`, doesn't even ship docker I think. `podman` should be available and work fine on RHEL7 (I haven't checked).
- Debian/Ubuntu/Arch then? :D

.Procedure

. Open a terminal and browse to the Apollo Voyager Server repo on your local machine.
+
. Create a Dockerfile, for example, `touch Dockerfile`.
+

> . Open the Docker file and add the following content, then save the changes:

Reviewer 2: suggestion:

combine 1, 2 and 3 into:

. Create a file named `Dockerfile` in the root directory of the Apollo Voyager Server repo, with the following contents:

> [id="creating-a-voyager-server-dockerfile-for-use-with-openshift-{context}"]

Reviewer: this (and the file name) are now inaccurate I think: it's not about creating a Dockerfile, but rather a container image.

+
. Open the Docker file and add the following content, then save the changes:
+
[source,dockerfile]
----
FROM node:8
WORKDIR /usr/src/app
# copy Node.js specific files
COPY package*.json ./
# copy application source file to the workdir
COPY index.js .
RUN npm install
# TCP port that application is listening on
EXPOSE 4000
CMD [ "node", "index.js" ]
----
+
. Build the container image and tag it:
+
[source,bash]
----
$ docker build -t aerogear/showcase-server:latest
----
+
NOTE: Alternatively, you can use `Podman` to build a Docker container: `podman build -t aerogear/showcase-server:latest .`
+
. Once the process is complete, a success message is displayed:
[source,bash]
----
$ Successfully built 6490624e298b
$ Successfully tagged aerogear/showcase-server:latest
----

.Verification steps

For the example above, to verify if a container image has been created locally, execute `docker images` and a similar message is displayed:
[source,bash]
----
REPOSITORY                                 TAG                 IMAGE ID            CREATED             SIZE
aerogear/showcase-server                   latest              6490624e298b        3 hours ago         1.1GB
----

> .Additional resources

Reviewer: ```suggestion
```

Probably can remove this empty section also
